src/Chapter04/Table04.04.CheckForNull.cs

```diff
@@ -2,44 +2,98 @@ namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter04.Table04_04
 {
     public class Program
     {
-        public static void EqualityCheckForNull(string? uri = null)
+        public static void PatternMatchingIsNull(string? uri = null!)
         {
-            if (uri != null)
+            // 1.
+            if (uri is null) 
+            {
+                Console.WriteLine(
+                    "Uri is null");
+            }
+            else
+            {
+                Console.WriteLine(
+                    $"Uri is: {uri}");
+            }
+        }
+        public static void PatternMatchingIsNotNull(string? uri = null)
+        {
+            // 2.
+            if (uri is not null)
             {
-                System.Console.WriteLine(
-                    $"Uri is: { uri }");
+                Console.WriteLine(
+                    $"Uri is: {uri}");
             }
-            else // (uri == null)
+            else
             {
-                System.Console.WriteLine(
+                Console.WriteLine(
                     "Uri is null");
             }
         }
-        public static void PatternMatchingCheckForNull(string? uri = null)
+        public static void EqualityInEqualityCheckForNull(string? uri = null)
         {
+            // 3.
+            if (uri == null)
+            {
+                Console.WriteLine(
+                    "Uri is null");
+            }
+
             if (uri != null)
             {
-                System.Console.WriteLine(
-                    $"Uri is: { uri }");
+                Console.WriteLine(
+                    $"Uri is: {uri}");
             }
-            else // (uri == null)
+
+        }
+        public static void IsObject(string? uri = null!)
+        {
+            // 4.
+            int number = 0;
+            if ((uri is object)
+                // Warning CS0183: The given
+                // expression is always not
+                // null.
```

Comment:
Github actions seems to suggest that the error message is "The given expression is always of the provided ('object') type", is maybe this error message slightly off?